tests/test_cluster_config_has_changed.py

```diff
@@ -1,23 +1,29 @@
+from pathlib import Path
+from typing import Iterator
+
 import nagiosplugin
+import pytest
 from click.testing import CliRunner
-from pytest_mock import MockerFixture
 
 from check_patroni.cli import main
 
-from .tools import here, my_mock
+from . import PatroniAPI
+
+
+@pytest.fixture(scope="module", autouse=True)
```

Comment:
Did you use this design to separate the setup from the tests proper ? (and avoid code duplication in some cases)

Reply:
Not really to separate the setup, although this is nice, rather to avoid code duplication as all test cases in the module use the same setup. The `autouse=True` avoids `@pytest.mark.usefixtures(...)` as used in other modules. The `scope="module"` avoids re-running the fixture for each test case, which is not needed as the state of the `PatroniAPI` object does not change between test cases.